thirdparty/faiss/faiss/impl/AdditiveQuantizer.cpp

```diff
@@ -322,7 +322,8 @@ void AdditiveQuantizer::compute_centroid_norms(float* norms) const {
 #pragma omp for
         for (int64_t i = 0; i < ntotal; i++) {
             decode_64bit(i, tmp.data());
-            norms[i] = fvec_norm_L2sqr(tmp.data(), d);
+            float norm = fvec_norm_L2sqr(tmp.data(), d);
+            norms[i] = (norm == 0.0 ? 1.0 : norm);
```

Comment:
no, because it uses norm for distance = x^2 - 2xy + y^2 calculation, not for a cosine-related code